src/test/java/wolox/training/controllers/BookControllerTests.java

```diff
@@ -0,0 +1,233 @@
+package wolox.training.controllers;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import wolox.training.exceptions.BookNotFoundException;
+import wolox.training.exceptions.DatabaseException;
+import wolox.training.exceptions.IdMismatchException;
+import wolox.training.models.Book;
+import wolox.training.repositories.BookRepository;
+
+@RunWith(SpringRunner.class)
+@WebMvcTest(BookController.class)
+class BookControllerTests {
+
+	@Autowired
+	private MockMvc mvc;
+	
+	@MockBean
+	private BookRepository bookRepository;
+
+	private final ObjectMapper mapper = new ObjectMapper();
+
+	private final String apiURL = "/api/books";
+	private final int bookId = 1;
+	private final int nonExistingBookId = 500;
+	private Book bookWithErrors;
+	private Book bookToUpdate;
+	private Book book;
+
+	@BeforeEach
+	public void setUp() {
+		mapper.configure(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS, false);
+	}
+
+	@BeforeEach
+	public void init() {
+
+		this.book = new Book();
+		this.book.setImage("http://someurl.com");
+		this.book.setGenre("Book genre");
+		this.book.setPages(500);
+		this.book.setPublisher("Some publisher");
+		this.book.setTitle("Book title");
+		this.book.setSubtitle("Book subtitle");
+		this.book.setYear("2000");
+		this.book.setIsbn("Some isbn");
+		this.book.setAuthor("Book author");
```

Comment:
you can extract this to another class to not repeat so much code, for example a `BookFactory` class